package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -222,6 +238,9 @@ define([
               paramDefn.autoSubmitUI = autoSubmit;
             }
 
+            if (this._oldParameterDefinition == null) {
+              this.extractDefaultValues(paramDefn);
+            }
```

Comment:
I think that logic is an error waiting to happen. Consider the following flow:

(1) Intiial call: oldParamDef == null. ; current paramDefn updated to hold the extracted default values.
(2) Then a second, server validating update call to server: 

paramDefnCallback will be invoked from fetchParameterDefinition. 
this.oldParamDefn is no longer null, but paramDefn is a parsed object that has been produced from the XML. Your previously stored default-values were on a different paramDefn object, and are now lost. As "oldParameterDefinition" is not null, the "extractDefaultValues" is not called (and calling it now would give you the current selection, not sensible default values).

It might be wise to store the default-values separately from the paramDefn object. Personally I;d think there is nothing wrong just storing it as a simple map, keyed by parameter-name, here on the "reportviewer-prompt" (ie "this") instead.